tests/unit/test_control.py

```diff
@@ -24,43 +23,24 @@ def test_passing_host(self):
         (-1, [{"status": {"ready": False}}], 0, 0),
     ])
     def test_create_index_with_timeout(self, mocker, timeout_value, describe_index_responses, expected_describe_index_calls, expected_sleep_calls):
-        mocker.patch.object(IndexOperationsApi, 'describe_index', side_effect=describe_index_responses)
```

Comment:
The IndexOperationsApi class generated by openapi 5.2.8 can't be mocked in the normal way because almost all of its methods are attributes defined at runtime from within __init__(). I'm not a mocker guru by any means, but the only way I could get this working was to grab the instance of IndexOperationsApi being used by the pinecone client and monkey patch that directly. 🤦